### ISSUE TITLE
SPM : Missing required modules: 'Realm.Private', 'Realm'

### ISSUE THREAD

**Issue by omokagbo** [realm/realm-swift#7266]
## Goals

To be able to use realm in my project by linking it via SPM.

## Expected Results

To be able to run Realm successfully without any errors

## Actual Results

Received build error: Missing required modules: 'Realm.Private', 'Realm' and could not make use of Realm

## Steps for others to Reproduce

 - include Realm/RealmSwift in a project via SPM
 - try importing Realm or using it any of its calls

## Code Sample

I could not use it to do anything. After installing via SPM and calling the import function, it threw up errors.

## Version of Realm and Tooling

Realm framework version: 10.7.6

Xcode version: 12.5

iOS/OSX version: 14

Dependency manager + version: SPM 5.3

**Comment by ericjordanmossman** [comment]
Yeah this does look similar #6301 (as you mentioned on that thread). I'm still not 100% sure what causes this, though. 
Let's keep 6301 closed and 7266 open, since this report is for 10.7.6.

And just to confirm -- are you able to import earlier versions of realm with SPM?

**Comment by omokagbo** [comment]
Same thing happens. The error kind of goes and comes back from what I observed. I'm thinking it might be a problem with xcode or something.

**Comment by ericjordanmossman** [comment]
And changing the [other swift flags](https://github.com/realm/realm-cocoa/issues/6301#issuecomment-614225989), no longer works with 10.7.6?

**Comment by paulthorsteinson** [comment]
I had no issues having realm as a dependency in a swift package that uses swift.  I also had no issues using realm in my project's main target. I did however have issues using realm in a secondary target of my project that worked fine when I had realm as a pod.  I tried adding the other swift flags and that did not fix the issue.  Ideally one would just add the realm package and it would work without having to change build settings if possible.

**Comment by paulthorsteinson** [comment]
In my case it was not adding [other swift flags](https://github.com/realm/realm-cocoa/issues/6301#issuecomment-614225989) that solved it, but rather I realized that when I added the package that in turn had a realm dependancy, the package was only added by default to one target.  I had to add the framework to each of my other targets for it to work

**Comment by jsflax** [comment]
This sounds like red herring error when trying to use Realm from targets it wasn't being linked to.

**Comment by syntakks** [comment]
Still having this error. I have a main workspace and an embedded framework. 
I used SPM to install Realm "up to next major" 10.19.0 (This ends up giving me 10.22.0)
In my framework I go to the Target > Build Phases > Link Binary with Libraries and Add the Realm libraries. 
Clean and Build and I get failure for symbols in RealmCore.

Don't think it's a red herring. I installed Alamofire exactly the same way and it has no problems? 

I'll start with my steps from scratch. 

### In Main Project 
1. File > Add Packages > (enter url: https://github.com/realm/realm-swift.git)
2. Up to Next Major Version 10.19.0 < 6.0.0 (There is a bug with how Xcode is showing the version so you have to increase it.)
3. Add to project: (My Main Project)
4. Add Package
5. Select Realm and RealmSwift Checkboxes and select my main target in "Add to Target" dropdown"
6. Packages are added and I see some warnings right away:
    realm-swift
    Invalid Exclude `path/to/this/thing/realm-swift/tools': File not found.
7. Clean and build with no errors.
8. Add test import of "RealmSwift" in AppDelegate and build/ run with no issues. Realm is working.

### Embedded Framework (Which I use as a way to separate server versions of my codebase.)
1. SubProject > Target > Build Phases > Link Binary With Libraries > "+" to add (Alamofire in here and works)
2. Add other > Add Package Dependency 
3. Select Realm Swift > (Up to Major Version 10.19.0 > 6.0.0) >Add to project "my subproject" > Add Package.
4. Check both boxes for Realm and RealmSwift > Add Package
5. Build and Run
6. Build Fails at "Package Realm Database" 
![image](https://user-images.githubusercontent.com/23320543/154384832-e0b37884-608f-41eb-96b3-e97049598b41.png)

**Comment by tgoyne** [comment]
That looks like we're not setting the linker flags for the Core target correctly. When only one target depends on Realm everything is built as a static library and Core's linker flags are ignored, but once there's multiple targets which depend on it SPM will build all the libraries as dynamic libraries and it does matter.

**Comment by andreaagudo3** [comment]
Any news on this? It is happening to me now in Xcode 14 beta.

**Comment by bmunkholm** [comment]
@andreaagudo3 Could you please create a new issue with all details on how to reproduce please?

**Comment by pitoneux** [comment]
Did we open a new issue with this ? Any workaround ? I'm having this issue with xcode 14.0 and 14.1 beta2

**Comment by markus-mohemian** [comment]
Had the same issue. Solved it by linking the binary targets under `<test-target> -> Build Phases -> Link Binary with Libraries`. There I added Realm and RealmSwift and then it built.

**Comment by ssup1** [comment]
> Had the same issue. Solved it by linking the binary targets under `<test-target> -> Build Phases -> Link Binary with Libraries`. There I added Realm and RealmSwift and then it built.

It doesn't work for me :(
Do we have a solution for this issue?

**Comment by harounhajem** [comment]
This is still an issue

**Comment by cutiko** [comment]
Documentation seems to be missing an appropriate explanation for the needed step. Documentation says:

https://www.mongodb.com/docs/realm/sdk/swift/test-and-debug/#test-targets:~:text=Since%20you%27re%20using%20Realm%20as%20a%20dynamic%20framework%2C%20you%27ll%20need%20to%20make%20sure%20your%20unit%20test%20target%20can%20find%20Realm.%20Add%20the%20parent%20path%20to%20RealmSwift.framework%20to%20your%20unit%20test%27s%20%22Framework%20Search%20Paths%22.

Which is actually a whole thing to do: https://stackoverflow.com/a/31897969/4017501

Leaving it here in case someone else finds the same problem since this was the first result on Google.
